python/setup.py

```diff
@@ -63,22 +64,40 @@ def get_cuda_version_from_header(cuda_include_dir):
 # valid symbols for specific version of CUDA.
 
 cwd = os.getcwd()
-preprocess_files = ["gpu.pxd"]
-supported_cuda_versions = {"10.1", "10.2", "11.0"}
-
-for file_p in preprocess_files:
-    pxi_file = ".".join(file_p.split(".")[:-1])
-    pxi_file = pxi_file + ".pxi"
-
-    if CUDA_VERSION in supported_cuda_versions:
-        shutil.copyfile(
-            os.path.join(cwd, "rmm/_cuda", CUDA_VERSION, pxi_file),
-            os.path.join(cwd, "rmm/_cuda", file_p),
+files_to_preprocess = ["gpu.pxd"]
+
+# The .pxi file is unchanged between some CUDA versions
+# (e.g., 11.0 & 11.1), so we keep only a single copy
+# of it
+cuda_version_to_pxi_dir = {
+    "10.1": "10.1",
+    "10.2": "10.2",
+    "11.0": "11.x",
+    "11.1": "11.x",
+    "11.2": "11.x",
+}
+
+for pxd_basename in files_to_preprocess:
+    pxi_basename = os.path.splitext(pxd_basename)[0] + ".pxi"
+    if CUDA_VERSION in cuda_version_to_pxi_dir:
+        pxi_pathname = os.path.join(
+            cwd,
+            "rmm/_cuda",
+            cuda_version_to_pxi_dir[CUDA_VERSION],
+            pxi_basename,
         )
+        pxd_pathname = os.path.join(cwd, "rmm/_cuda", pxd_basename)
+        try:
+            if filecmp.cmp(pxi_pathname, pxd_pathname):
+                # files are the same, no need to copy
+                continue
+        except FileNotFoundError:
+            # pxd_pathname doesn't exist yet
+            pass
+        shutil.copyfile(pxi_pathname, pxd_pathname)
```

Comment:
Can we move the cuda version check outside of the loop and invert it to reduce nesting?
```python
if CUDA_VERSION not in cuda_version_to_pxi_dir:
    raise TypeError(f"{CUDA_VERSION} is not supported.")
```

Reply:
That would mean we always check, regardless of how many files we have to preprocess, so that might need to be accounted for.

Reply:
Agreed that this is low hanging fruit to fix and we may as well tackle it now.

Reply:
Woops, this merged before fixing this. Will raise an issue to tackle it in a follow up.

Reply:
Ah sorry. I'll put in one tomorrow.